examples/user_guide/Templates.ipynb

```diff
@@ -0,0 +1,191 @@
+    "\n",
+    "## What is a template?\n",
+    "\n",
+    "A template is defined using the [jinja2](http://jinja.pocoo.org/docs/) templating language which makes it straightforward to extend the default template in various ways or even replace it entirely. Before modifying the default template let us take a look at it in its entirety:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```html\n",
+    "<!DOCTYPE html>\n",
+    "<html lang=\"en\">\n",
+    "{% block head %}\n",
+    "<head>\n",
+    "    {% block inner_head %}\n",
+    "    <meta charset=\"utf-8\">\n",
+    "    <title>{% block title %}{{ title | e if title else \"Panel App\" }}{% endblock %}</title>\n",
+    "    {% block preamble %}{% endblock %}\n",
+    "    {% block resources %}\n",
+    "        {% block css_resources %}\n",
+    "        {{ bokeh_css | indent(8) if bokeh_css }}\n",
+    "        {% endblock %}\n",
+    "        {% block js_resources %}\n",
+    "        {{ bokeh_js | indent(8) if bokeh_js }}\n",
+    "        {% endblock %}\n",
+    "    {% endblock %}\n",
+    "    {% block postamble %}{% endblock %}\n",
+    "    {% endblock %}\n",
+    "</head>\n",
+    "{% endblock %}\n",
+    "{% block body %}\n",
+    "<body>\n",
+    "    {% block inner_body %}\n",
+    "    {% block contents %}\n",
+    "        {% for doc in docs %}\n",
+    "        {{ embed(doc) if doc.elementid }}\n",
+    "        {% for root in doc.roots %}\n",
+    "            {{ embed(root) | indent(10) }}\n",
+    "        {% endfor %}\n",
+    "        {% endfor %}\n",
+    "    {% endblock %}\n",
+    "    {{ plot_script | indent(8) }}\n",
+    "    {% endblock %}\n",
+    "</body>\n",
+    "{% endblock %}\n",
+    "</html>\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As we can see the template defines a number of custom blocks, which can be overriden in by extending this default template."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Using custom templates"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import panel as pn\n",
+    "import holoviews as hv\n",
+    "\n",
+    "pn.extension()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once we have loaded panel we can start defining a custom template, as mentioned before it is usually easiest to simply extend an existing template by overriding certain blocks. To begin with we start by using `{% extends base %}` to declare that we are merely extending an existing template rather than defining a whole new one, otherwise we would have to repeat the entire header sections of the full template to ensure all the appropriate resources are loaded.\n",
```

Comment:
```suggestion
    "Once we have loaded Panel we can start defining a custom template. As mentioned before, it is usually easiest to simply extend an existing template by overriding certain blocks. To begin with we start by using `{% extends base %}` to declare that we are merely extending an existing template rather than defining a whole new one; otherwise we would have to repeat the entire header sections of the full template to ensure all the appropriate resources are loaded.\n",
```